sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -631,18 +631,20 @@ class ExecChecks private(
     val plan = meta.wrapped.asInstanceOf[SparkPlan]
     val allowDecimal = meta.conf.decimalTypeEnabled
 
-    val unsupportedOutputTypes = plan.output
-      .filterNot(attr => check.isSupportedByPlugin(attr.dataType, allowDecimal))
-      .toSet
+    val unsupportedOutputTypes = plan.output.collect {
```

Comment:
I realize that returning the attribute instead of the type was a bug, but I have actually found it somewhat informative to have the name of the attribute in there as well.  Perhaps we could change `unsupportedOutputTypes` to be something more like `Map[DataType, Array[String]]` so we can output something more like.

```
!Exec cannot run on GPU because nested types in grouping expressions are not supported; unsupported data types in input: StructType(StructField(a,IntegerType,true), StructField(b,IntegerType,true)) [a#0]; unsupported data types in output: StructType(StructField(a,IntegerType,true), StructField(b,IntegerType,true)) [a#0]
```